liota/dccs/iotcc.py

```diff
@@ -170,11 +169,11 @@ def on_response(msg):
 
         self.comms.send(json.dumps(self._unregistration(self.next_id(), entity_obj.ref_entity)))
         on_response(self.recv_msg_queue.get(True, 20))
-        self.remove_reg_entity_details(entity_obj.ref_entity.name, entity_obj.reg_entity_id)
         if entity_obj.ref_entity.entity_type != "HelixGateway":
             self.store_device_info(entity_obj.reg_entity_id, entity_obj.ref_entity.name,
                                    entity_obj.ref_entity.entity_type, None, True)
         else:
+            self.remove_reg_entity_details(entity_obj.ref_entity.name, entity_obj.reg_entity_id)
```

Comment:
Can you please mention the reason to move this call in else loop for my understanding?

Reply:
now we only save edge system info in iotcc.json, no device info any more.
so should not do it devices. (!= "HelixGateway" is for devices).
If do, it will be in vain, and print some "cannot find xxx device info" warning/error, which is not what we want.

Reply:
Okay and but do we support any entity with device_type as none, I think we can get rid of else loop?

Reply:
No, we need to generate uuid.json file with "remove": true, so need that if...else to do that.

Reply:
I get the point but do we also store the information if the Helixgateway is deleted as the else is currently handling that.

Reply:
I think we still need it when we only delete some devices with helix gateway is still alive (still need to call store_device_info with specific parameters.)